test/test_ripper_tags.rb

```diff
@@ -76,6 +76,25 @@ def imethod5
     ], tags.map{ |t| t[:full_name] }
   end
 
+  def test_nested_constant_definitions
+    tags = extract(<<-EOC)
+      A = [
+        B = 1,
+        C = 2
+      ]
+
+      D = {
+        E = 3 => F = 4,
+        x: G = 5
```

Comment:
Do we need to support `F` and `G` here, i.e. parsing assignment on the right side of the `=>` hash syntax? Does anyone define constants that way?

Reply:
I don't think so, but it falls out "for free" from the use of `on_assoc_new` (and it would actually be harder to support nested hashes if we tried to only handle definitions in keys)

Reply:
> but it falls out "for free"

Even though we get is basically for "free", that is not reason enough to test for and support a feature in the library. If we don't need a specific feature, let's not test for it. Keep in mind that every test that we add here, I have to maintain for months/years over future releases of Ruby and ripper-tags itself. Let's keep the scope of the feature focused on just what was the initial goal.